### PR TITLE
Cleanup code

### PR DESCRIPTION
**Pull request by paluchasz**
# Description

Refactoring code - make sure it is working and splitting into modules
# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have updated the changelog with a short summary of this update
- [ ] I have added tests that prove my feature works

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -8,15 +8,20 @@ packages = [{include = "sentiment_analysis"}]
...
 scikit-learn = "^1.2.0"
+python-dotenv = "0.21.0"
+pydantic = "^1.10.2"
 
 [tool.poetry.group.dev.dependencies]
 pytest = "^7.2.0"
 plotly = "^5.11.0"
 pre-commit = "^2.20.0"
+isort = "^5.11.4"
+black = "^22.12.0"
```
Comment (felixcs1) [inline pyproject.toml:24]: did these get added automatically or are you adding them for a reason?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -38,8 +38,8 @@ repos:
...
-  - repo: https://github.com/econchick/interrogate
-    rev: 1.5.0
-    hooks:
-      - id: interrogate
-        args: [--fail-under=70, --verbose, --generate-badge=./docs/_static]
+  # - repo: https://github.com/econchick/interrogate
+  #   rev: 1.5.0
+  #   hooks:
+  #     - id: interrogate
+  #       args: [--fail-under=70, --verbose, --generate-badge=./docs/_static]
```
Comment (felixcs1) [inline .pre-commit-config.yaml]: So I agree that the badge generation is clunky, so maybe we just keep the hook in but remove the `--generate-badge` arg and delete the badge and the reference in the Readme. [Heres](https://interrogate.readthedocs.io/en/latest/#configuration) a set of configuration for the package. I think maybe to make it ignore __init__ files we would want `ignore-module = true`..... not certain, just seems the most likely

File: sentiment_analysis/scripts/train.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+
+class EnvVars(BaseSettings):
+    """Pydantic data class to hold environment variables"""
+
+    TRAIN_DATA_PATH: Path
+    EPOCHS: int = 1
+    BATCH_SIZE: int = 64 if GPU_AVAILABLE else 1
+    VAL_BATCH_SIZE: int = 16 if GPU_AVAILABLE else 1
+    SAVE_MODEL_DIR: Optional[Path] = None
```
Comment (felixcs1) [inline sentiment_analysis/scripts/train.py:26]: Not for this PR but we should move all this stuf into a config file (json/yaml or similar)

File: sentiment_analysis/scripts/train.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+import numpy as np
+import tensorflow as tf
+from dotenv import load_dotenv
+from pydantic import BaseSettings
+from sklearn.model_selection import train_test_split
+from transformers import DistilBertTokenizer, TFDistilBertForSequenceClassification
+
+GPU_AVAILABLE = tf.test.is_gpu_available()
+
+load_dotenv(".env")  # need to run script from root of project to read this
```
Comment (felixcs1) [inline sentiment_analysis/scripts/train.py:16]: should this `.env` file be committed? I dont see it, does it have vars that I will need to run the code?

**Review by felixcs1 — COMMENTED** [review]
Can you give a minimal code example in `main.py` file in the root so that it can be run from there with `python main.py`, something like:

```
from sentiment_analysis.scripts.train import Trainer
trainer = Trainer()
trainer.train()
```
Or however you would call it with how the code is currently.

Also can you add a line to the CHANGELOG.md

File: sentiment_analysis/scripts/train.py (hunk 3 above)
Comment (paluchasz) [reply sentiment_analysis/scripts/train.py:26]: Sure lets discuss that, I was using CLI args so far but saw my colleague use environment variables and pydantic classes a lot so thought I will try that

File: sentiment_analysis/scripts/train.py (hunk 4 above)
Comment (paluchasz) [reply sentiment_analysis/scripts/train.py:16]: I think generally not as it contains paths specific to my system and you will have your own paths

File: pyproject.toml (hunk 1 above)
Comment (paluchasz) [reply pyproject.toml:24]: I added these yes. It is so I can run them myself. I will also add mypy and flake8 here later. Also you can specify extra configurations for them. Then if you have a tool like mypy in Vscode/pycharm it should read this configuration too

File: sentiment_analysis/scripts/train.py (hunk 4 above)
Comment (felixcs1) [reply sentiment_analysis/scripts/train.py:16]: OK I think the paths for storing data and models should be generic i.e data should be in say `./data/` folder and models, encoders etc in `./artifacts/` or similar, where it is assumed these are relative paths from the project root. We should aim for zero user specific configuration i.e no absolute paths specific to someones machine

Comment (felixcs1) [reply sentiment_analysis/scripts/train.py:16]: Also the training code should check for these folders and create them if they don't exist (out of scope for this PR though)

File: pyproject.toml (hunk 1 above)
Comment (felixcs1) [reply pyproject.toml:24]: Cool, I guess if they are intended to be run not part of the commit hooks we should have script to run them, we shouldn't expect a contributer to run each package individually (i.e in this case me, but we're making it generally as project that should be easy to pick up and develop on with ease).

Comment (felixcs1) [reply pyproject.toml:24]: could be in a `scripts` folder

File: sentiment_analysis/scripts/train.py (hunk 4 above)
Comment (paluchasz) [reply sentiment_analysis/scripts/train.py:16]: I am making a sample folder and env file atm, and working on them being absolute yes

Comment (felixcs1) [reply sentiment_analysis/scripts/train.py:16]: though personally I prefer using pre commit for this task, if you want to run the commit hooks without actually commit you can run 
```
pre-commit run --all-files
```
the reason I prefer this is that then we have a single file (precommit config) which defines what we are using for code styling/linting and a single place to call it from.

Comment (felixcs1) [reply sentiment_analysis/scripts/train.py:16]: Also if you want to run something specifically like black you can do 
```
pre-commit run black  --all-files
```

**Review by felixcs1 — APPROVED** [review]

**Pull request merged**